doc/neps/tools/build_index.py

```diff
@@ -38,7 +38,7 @@ def nep_metadata():
 
             # The title should be the first line after a line containing only
             # * or = signs.
-            for i, line in enumerate(lines[:-1]):
+            for line in lines[:-1]:
```

Comment:
Hmm,  ` i` is used in line 48. I don't know if this should be classed as a ruff bug, or obscure programming. Apparently the `i` is not local to the `for` statement.

Reply:
I think this is a ruff bug.

Reply:
Indeed, `enumerate` shouldn't have been removed here, I'll revert.

This is not necessarily a ruff bug, in the sense that I had run it with `--fix --unsafe-fixes`. I hadn't noticed the later use of `i` when reviewing the changes.